channeldb/channel.go

```diff
@@ -166,6 +171,12 @@ func (c ChannelType) IsTweakless() bool {
 	return c&SingleFunderTweaklessBit == SingleFunderTweaklessBit
 }
 
+// HasFundingTx returns true if this channel type is one that has a funding
+// transaction stored locally.
+func (c ChannelType) HasFundingTx() bool {
```

Comment:
If the bit _isn't_ set, then that means we _have_ the funding transaction. If it's set, then it means we don't. As written, if we're about to attempt to handle the funding transaction, we also check if we're the initiator or not.

Reply:
> ... we also check if we're the initiator or not.

Yeah, this needs to be done which is why I think `HasFundingTx` is a dangerous name, as there is no guarantee we have it even though it returns true.

Reply:
I guess it makes sense since it is the _channel type_ that has the funding tx, not the _channel_. I'm okay with keeping the name, but adding to the godoc that it is only the initiator that has the tx strored locally, but I agree with @wpaulino that a more descriptive name could lessen the confusion.